webapp/sources/ldap-inventory/inventory-api/src/main/scala/com/normation/inventory/domain/DataTypes.scala

```diff
@@ -53,67 +53,71 @@ import zio._
 import zio.syntax._
 
 /**
- * A file that contains all the simple data types, like Version,
- * MemorySize, Manufacturer, etc.
+ * A file that contains all the simple data types, like Version, MemorySize, Manufacturer, etc.
  */
 
 /**
- * A simple class to denote a manufacturer
- * TODO : Should be merge with SoftwareEditor
+ * A simple class to denote a manufacturer TODO : Should be merge with SoftwareEditor
  */
-final case class Manufacturer(name:String) extends AnyVal
+final case class Manufacturer(name: String) extends AnyVal
 
 /**
  * A simple class to denote a software editor
  */
-final case class SoftwareEditor(val name:String) extends AnyVal
+final case class SoftwareEditor(val name: String) extends AnyVal
 
 sealed trait SecurityToken {
-  def key : String
+  def key: String
 }
 
 case object SecurityToken {
-  def kind(token : SecurityToken) = {
+  def kind(token: SecurityToken) =
     token match {
       case _: PublicKey   => PublicKey.kind
       case _: Certificate => Certificate.kind
     }
-  }
 
-  def parseCertificate(cert: Certificate): IO[InventoryError, (java.security.PublicKey, List[(String, String)])] = {
+  def parseCertificate(cert: Certificate): IO[InventoryError, (java.security.PublicKey, List[(String, String)])] =
     cert.cert.flatMap { ch =>
       IO.effect {
-        val c = new JcaX509CertificateConverter().getCertificate( ch )
-        val dn = ch.getSubject.getRDNs.flatMap(_.getTypesAndValues.flatMap(tv => (tv.getType.toString, tv.getValue.toString) :: Nil)).toList
+        val c = new JcaX509CertificateConverter().getCertificate(ch)
+        val dn = ch.getSubject.getRDNs
+          .flatMap(_.getTypesAndValues.flatMap(tv => (tv.getType.toString, tv.getValue.toString) :: Nil))
+          .toList
         (c.getPublicKey, dn)
       }.mapError { ex =>
         CryptoEx(s"Error when trying to parse agent certificate information", ex)
       }
     }
-  }
 
   def checkCertificateSubject(nodeId: NodeId, subject: List[(String, String)]): IO[InventoryError, Unit] = {
-    //format subject
+    // format subject
     def formatSubject(list: List[(String, String)]) = list.map(kv => s"${kv._1}=${kv._2}").mkString(",")
 
     // in rudder, we ensure that at list one (k,v) pair is "UID = the node id". If missing, it's an error
-    subject.find { case (k,v) => k == ParsedSecurityToken.nodeidOID } match {
-      case None        => InventoryError.SecurityToken(s"Certificate subject doesn't contain node ID in 'UID' attribute: ${formatSubject(subject)}").fail
-      case Some((k,v)) =>
-        if(v.trim.equalsIgnoreCase(nodeId.value)) {
+    subject.find { case (k, v) => k == ParsedSecurityToken.nodeidOID } match {
+      case None =>
+        InventoryError
+          .SecurityToken(s"Certificate subject doesn't contain node ID in 'UID' attribute: ${formatSubject(subject)}")
+          .fail
```

Comment:
i'm not totally sure of this one: it's more readable, but it adds lines, so making skimming through the code harder.

Reply:
we will need to see if after a couple weeks it's a net gain or not